agent-tooling/src/main/java/io/opentelemetry/auto/tooling/matcher/GlobalIgnoresMatcher.java

```diff
@@ -63,110 +63,154 @@ private GlobalIgnoresMatcher(final boolean skipAdditionalLibraryMatcher) {
+        break;
+      case 'r' - 'a':
+        break;
+      case 's' - 'a':
+        if (name.startsWith("sun.")) {
+          return !name.startsWith("sun.net.www.protocol.")
+              && !name.startsWith("sun.rmi.server")
+              && !name.startsWith("sun.rmi.transport")
+              && !name.equals("sun.net.www.http.HttpClient");
+        }
+        break;
+      default:
     }
 
-    if (name.startsWith("java.")) {
-      if (name.equals("java.net.URL") || name.equals("java.net.HttpURLConnection")) {
-        return false;
+    final int firstDollar = name.indexOf('$');
+    if (firstDollar > -1) {
+      // clojure class patterns
+      if (name.startsWith("loader__", firstDollar + 1)) {
+        return true;
       }
-      if (name.startsWith("java.rmi.") || name.startsWith("java.util.concurrent.")) {
-        return false;
-      }
-      // Concurrent instrumentation modifies the structure of
-      // Cleaner class incompatibly with java9+ modules.
-      // Working around until a long-term fix for modules can be
-      // put in place.
-      if (name.startsWith("java.util.logging.")
-          && !name.equals("java.util.logging.LogManager$Cleaner")) {
-        return false;
+      int dollar = firstDollar;
+      while (dollar > -1) {
+        if (name.startsWith("fn__", dollar + 1) || name.startsWith("reify__", dollar + 1)) {
+          return true;
+        }
+        dollar = name.indexOf('$', dollar + 1);
       }
 
-      return true;
-    }
-
-    if (name.startsWith("com.sun.")) {
-      if (name.startsWith("com.sun.messaging.") || name.startsWith("com.sun.jersey.api.client")) {
-        return false;
+      if (name.contains("$JaxbAccessor")
+          || name.contains("CGLIB$$")
+          || name.contains("$__sisu")
+          || name.contains("$$EnhancerByProxool$$")
+          || name.startsWith("org.springframework.core.$Proxy")) {
+        return true;
       }
-
-      return true;
     }
-
-    if (name.startsWith("sun.")) {
-      if (name.startsWith("sun.net.www.protocol.")
-          || name.startsWith("sun.rmi.server")
-          || name.startsWith("sun.rmi.transport")
-          || name.equals("sun.net.www.http.HttpClient")) {
-        return false;
-      }
-
-      return true;
-    }
-
-    if (name.startsWith("org.slf4j.")) {
-      if (name.equals("org.slf4j.MDC")) {
-        return false;
-      }
-
-      return true;
-    }
-
-    if (name.contains("$JaxbAccessor")
-        || name.contains("CGLIB$$")
-        || name.contains("javassist")
-        || name.contains(".asm.")
-        || name.contains("$__sisu")
-        || name.contains("$$EnhancerByProxool$$")
-        || name.startsWith("org.springframework.core.$Proxy")) {
-      return true;
-    }
-
-    if (COM_MCHANGE_PROXY.matcher(name).matches()) {
-      return true;
-    }
-
-    if (!skipAdditionalLibraryMatcher && additionalLibraryIgnoreMatcher.matches(target)) {
+    if (name.contains("javassist") || name.contains(".asm.")) {
```

Comment:
Aren't these so broad they'll result in false positives? I notice it's not a new change but noticed it now, especially since I had the same question about the clojure patterns, which also seem like they have a higher chance of colliding with a user class than seems safe.

Reply:
Possibly, but they aren't new, they just moved.

Reply:
Opened #769 to review all global ignores